server/src/services/semanticHighlight/onSemanticTokensFull.ts

```diff
@@ -0,0 +1,159 @@
+          );
+          return {
+            status: "internal_error",
+            result: emptyResponse,
+          };
+        }
+
+        try {
+          // Parse using slang
+          span = transaction.startChild({ op: "slang-parsing" });
+          const language = new Language(solcVersion!);
+
+          const parseOutput = language.parse(
+            ProductionKind.SourceUnit,
+            document.getText()
+          );
+
+          const parseTree: SlangNode = parseOutput.parseTree;
+          span.finish();
+
+          // Register visitors
+          const builder = new SemanticTokensBuilder(document);
+
+          const visitors = [
+            new CustomTypeHighlighter(document, builder),
+            new FunctionDefinitionHighlighter(document, builder),
+            new FunctionCallHighlighter(document, builder),
+            new EventEmissionHighlighter(document, builder),
+            new EventDefinitionHighlighter(document, builder),
+            new ContractDefinitionHighlighter(document, builder),
+            new InterfaceDefinitionHighlighter(document, builder),
+            new StructDefinitionHighlighter(document, builder),
+            new UserDefinedValueTypeDefinitionHighlighter(document, builder),
+            new EnumDefinitionHighlighter(document, builder),
+            new ErrorDefinitionHighlighter(document, builder),
+            new LibraryDefinitionHighlighter(document, builder),
+          ];
+
+          // Visit the CST
+          const indexedVisitors: Dictionary<HighlightVisitor[]> = {};
+          const registeredTokenKinds: TokenKind[] = [];
+
+          for (const visitor of visitors) {
+            for (const tokenKind of visitor.tokenKinds) {
+              indexedVisitors[tokenKind] ||= [];
+              indexedVisitors[tokenKind].push(visitor);
+
+              if (!registeredTokenKinds.includes(tokenKind)) {
+                registeredTokenKinds.push(tokenKind);
+              }
+            }
+          }
+
+          const cursor: Cursor = parseTree.cursor;
+          let node: TokenNode;
+
+          span = transaction.startChild({ op: "walk-highlight-tokens" });
+          while (
+            (node = cursor.findTokenWithKind(registeredTokenKinds)) !== null
+          ) {
+            const nodeWrapper = {
+              kind: node.kind,
+              pathRuleNodes: cursor.pathRuleNodes,
+              text: node.text,
+              textRange: cursor.textRange,
+              type: node.type,
+            };
+
+            const registeredVisitors = indexedVisitors[nodeWrapper.kind];
+            for (const visitor of registeredVisitors) {
+              visitor.enter(nodeWrapper);
+            }
+
+            cursor.goToNext();
+          }
+          span.finish();
+
+          return { status: "ok", result: { data: builder.getTokenData() } };
+        } catch (error) {
+          logger.error(`Slang parsing error: ${error}`);
```

Comment:
I suggest specializing this `try/catch` to just Slang parsing, or updating the error title:

```suggestion
          logger.error(`Semantic Highlighting Error: ${error}`);
```

Reply:
This has been addressed